pkg/reconcilers/shared/reconciler.go

```diff
@@ -72,7 +78,7 @@ func (r *BaseReconciler) UpdateStatusForSyncingError(hcp *tenancyv1alpha1.Contro
 		// Requeue after 10 seconds, don't mark as failed
 		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 	}
-	return ctrl.Result{}, err
+	return ctrl.Result{}, e
```

Comment:
This change is relevant to all the calls on this func, and there are many. I think that this change is an improvement, but I suspect that @pdettori was thinking differently.
/cc @pdettori